pkg/helmexecutor/client.go

```diff
@@ -0,0 +1,107 @@
+// Package helmexecutor provides methods for executing helm commands as methods.
+// It is a wrapper around the helmclient of https://github.com/mittwald/go-helm-client.
+package helmexecutor
+
+import (
+	"errors"
+
+	helmclient "github.com/mittwald/go-helm-client"
+	"helm.sh/helm/v3/pkg/repo"
+	"k8s.io/client-go/rest"
+)
+
+// HelmExecutor provides an interface for implementing different helmclient types.
+type HelmExecutor interface {
+	// UpdateHelmRepos updates the list of chart repositories stored in the executors cache.
+	UpdateHelmRepos() error
+
+	// AddOrUpdateChartRepo adds a new or updates an existing chart repository.
+	AddOrUpdateChartRepo(repo.Entry) error
+
+	// InstallOrUpgradeHelmChart installs or upgrades a chart referenced by helmChart.
+	InstallOrUpgradeHelmChart(*helmclient.ChartSpec) error
+
+	// UninstallHelmRelease uninstalls the helmchart referenced by helmChart.
+	UninstallHelmRelease(*helmclient.ChartSpec) error
+}
+
+// HelmClientExecutor is a wrapper around github.com/mittwald/go-helm-client helmclient
+// implementing the HelmExecutor interface.
+type HelmClientExecutor struct {
+	RepositoryCache string
+	RepositoryConfig string
+	RestConfig *rest.Config
+}
+
+// UpdateHelmRepos implements the HelmExecutor interface.
+func (h *HelmClientExecutor) UpdateHelmRepos() error {
+	helmClient, err := helmclient.New(&helmclient.Options{
+		RepositoryCache:  h.RepositoryCache,
+		RepositoryConfig: h.RepositoryConfig,
+	})
+	if err != nil {
+		return err
+	}
+
+	return helmClient.UpdateChartRepos()
```

Comment:
It bugs me a little bit that the methods in this interface are named _similar_ to the Helm client, but not _exactly_ the same. For example, `UpdateHelmRepos` vs `UpdateChartRepos`, or `UninstallHelmRelease` vs `UninstallRelease`.

Maybe name them the same? That would also have the benefit that the helmClient itself would also implement the `HelmExecutor` interface, making them completely interchangeable.

Reply:
I'm currently rewriting the code to accomodate @jkmw 's changes, which will make sure that we don't need a wrapper.